### PR TITLE
Support for merge_commit_sha

### PR DESCRIPTION
**Pull request by recena**
`GHPullRequest` object now includes `merge_commit_sha`

@reviewbybees specially, @oleg-nenashev

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHPullRequest.java [hunk 1]
```diff
@@ -186,6 +189,15 @@ public int getChangedFiles() throws IOException {
     }
 
     /**
+     * FYI: https://developer.github.com/changes/2013-04-25-deprecating-merge-commit-sha
+     */
+    @Deprecated
+    public String getMergeCommitSha() throws IOException {
```
Comment (oleg-nenashev) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: What happens if the field disappears? Does the method return null in such case?

Comment (recena) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: When this happens, GitHub API v4 will be released and we should review the current status (compatibility).

File: src/main/java/org/kohsuke/github/GHPullRequest.java [hunk 2]
```diff
@@ -142,9 +145,9 @@ public PullRequest getPullRequest() {
     }
 
     //
-// details that are only available via get with ID
-//
-//
```
Comment (recena) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: Unrelated changes but...

File: src/main/java/org/kohsuke/github/GHPullRequest.java (hunk 1 above)
Comment (recena) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: @oleg-nenashev Any recommendation here to improve it?

Comment (recena) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: @oleg-nenashev I've checked that if your model has an attribute and is not mapped in the JSON, its `get()` method returns `null`. I think that is ok.

File: src/main/java/org/kohsuke/github/GHPullRequest.java (hunk 2 above)
Comment (oleg-nenashev) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: Makes sense to convert the comment to Javadoc then :)

**Comment by lanwen** [comment]
why you want to use deprecated field?

File: src/main/java/org/kohsuke/github/GHPullRequest.java (hunk 1 above)
Comment (amuniz) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: Right. [This line](https://github.com/recena/github-api/blob/b0789a7ce72b6e1773b47d4e778535f839863480/src/main/java/org/kohsuke/github/GitHub.java#L564) is preventing to fail on missing properties.

**Comment by KostyaSha** [comment]
@lanwen because it can be valid for some unknown time or on some ancient enterprise installations?

File: src/main/java/org/kohsuke/github/GHPullRequest.java [hunk 3]
```diff
@@ -186,6 +189,15 @@ public int getChangedFiles() throws IOException {
     }
 
     /**
+     * FYI: https://developer.github.com/changes/2013-04-25-deprecating-merge-commit-sha
```
Comment (KostyaSha) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: javadoc `@link`?

**Comment by amuniz** [comment]
:bee:

File: src/main/java/org/kohsuke/github/GHPullRequest.java (hunk 3 above)
Comment (lanwen) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: javadoc `@deprecated`

Comment (recena) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: Sure.

File: src/main/java/org/kohsuke/github/GHPullRequest.java (hunk 2 above)
Comment (recena) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: Really, it is not a method comment, only a note in the middle of the file.

File: src/main/java/org/kohsuke/github/GHPullRequest.java (hunk 1 above)
Comment (KostyaSha) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: @oleg-nenashev `@CheckForNull`?

**Comment by recena** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
:+1:

**Comment by jglick** [comment]
Does not make sense to introduce a deprecated API, especially as this will probably still working in v4.

**Comment by KostyaSha** [comment]
probably makes sense call variable in camel case according to java convention and use Json annotation.

**Comment by recena** [comment]
@jglick if you don't need to use it, simply ignore it. GH API V4 is not planned, there is not a date. While this happens this attribute is the only way to resolve some problems.

**Comment by recena** [comment]
@jglick make sense to include not documented behaviors? For example, retrieve `/merge` spec???

**Comment by recena** [comment]
@KostyaSha I agree but I tried to follow the current conventions in this project.

**Comment by recena** [comment]
@jglick I apologize. I understood wrong your comment.I should read twice before to reply. Specially, when I read english text.

**Comment by recena** [comment]
@jglick I'm going to remove the tag `@decrecated`.

**Comment by recena** [comment]
@jglick `@deprecated` annotations were removed.

**Comment by ndeloof** [comment]
LGTM

**Comment by recena** [comment]
@reviewbybees done

**Comment by jglick** [comment]
Belated :bee:.

I do not think we really want to be using this API. But I see no harm in exposing it via the library. It is up to the caller whether to rely on it or not. I presume that if and when the field is deleted from the JSON by GH, `getMergeCommitSha` will simply return null and that is that.

**Comment by recena** [comment]
@jglick If a field is removed from GitHub API, its getter asociated will return `null`.

**Pull request merged**